docs/bugs.json

```diff
@@ -1,4 +1,12 @@
 [
+    {
+        "name": "FreeFunctionRedefinition",
+        "summary": "Call to free function that has been redefined within the same source unit or via the import statement can result in undefined behavior.",
```

Comment:
Undefined because

- successive redefinition in single source unit returns the result of the first definition (see https://github.com/ethereum/solidity/blob/ad5fef714d6a3216f81e87e9009f297aad4e1710/test/libsolidity/semanticTests/freeFunctions/redefinition_mutability.sol)
- successive redefinition across two source unit returns the result of the second definition (see https://github.com/ethereum/solidity/blob/ad5fef714d6a3216f81e87e9009f297aad4e1710/test/libsolidity/semanticTests/multiSource/free_function_resolution_override_virtual.sol)

Reply:
This can be generalised as first redefinition in source unit is picked but that is confusing, so it is better to call this undefined behavior.

Reply:
I think "undefined behaviour" is too strong - it cannot modify storage if none of the functions modifies storage, for example.

Reply:
Okay, I understand. I worded it as a worst case scenario. What do you think of the following?

```
Call to free function that has been redefined within the same source unit or via the import statement invokes the first instance of the redefined function within the source unit."

Reply:
I would actually avoid the term "redefinition" and write something like "compiler does not detect the situation of two or more free functions of the same name and parameter types".